cmd/entc/entc.go

```diff
@@ -194,15 +194,27 @@ func createDir(target string) error {
 	if err := os.MkdirAll(target, os.ModePerm); err != nil {
 		return fmt.Errorf("creating schema directory: %w", err)
 	}
-	if target != defaultSchema {
-		return nil
-	}
-	if err := ioutil.WriteFile("ent/generate.go", []byte(genFile), 0644); err != nil {
+	if err := ioutil.WriteFile(target+"/../generate.go", []byte(getGenerateFileData(target)), 0644); err != nil {
 		return fmt.Errorf("creating generate.go file: %w", err)
 	}
 	return nil
 }
 
+// ent/generate.go file used for "go generate" command.
+func getGenerateFileData(target string) string {
+	pathArr := strings.Split(target, "/")
+
+	var sb strings.Builder
+	sb.Grow(32)
+
+	sb.WriteString("package ent\n\n")
+	sb.WriteString("//go:generate go run github.com/facebookincubator/ent/cmd/entc generate ")
+	sb.WriteString("./" + pathArr[len(pathArr)-1])
+	sb.WriteString("\n")
+
+	return sb.String()
+}
```

Comment:
Use `filepath` functions (as mentioned above) and return a `[]byte` because conversion above is redundant.

Reply:
Code updated in https://github.com/facebookincubator/ent/pull/585/commits/59dcd397fd020d5bbc111f6c26756f213a39dc78.

I'm not sure if there is a better way to convert `strings.Builder` to a `[]byte`.